internal/proto/authd/authd.pb.go

```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.35.1
-// 	protoc        v3.21.12
+// 	protoc        v4.23.4
 // source: authd.proto
 
 package authd
```

Comment:
> we've not just the authd protocol, but also the gdm and pam ones, that is are the gdm and proto namespaces

So that's also inconsistent. With your proposed change, we have the following packages containing protobuf definitions:
* `internal/proto/authd` -> contains the protobuf definitions for the PAM and NSS gRPC services
* `pam/internal/proto` -> contains a stage enum which doesn't seem to be used in any gRPC API (?) (why is this even defined in protobuf?)
* `pam/internal/gdm` -> contains protobuf message definitions which only seem to be used internally in the PAM module (why is this defined in protobuf?)

Apart from the fact that i wonder whether it makes sense at all to have the `gdm` and `pam` protobuf definitions, the naming is inconsistent.

Reply:
> pam/internal/proto -> contains a stage enum which doesn't seem to be used in any gRPC API (?) (why is this even defined in protobuf?)

It's used by the gdm one and to have a static and unique way to define the stage types. And ideally this should have to contain more things.

The naming was a bit unfortunate because we didn't find anything generic enough to cover something that was just a building block for other protos.

> pam/internal/gdm -> contains protobuf message definitions which only seem to be used internally in the PAM module (why is this defined in protobuf?)

Protobufs are just definitions of protocols to generate structs and code easily and consistently for multiple languages. In this case, it uses protobuf JSON generator (way more consistent than native go one) and on the shell side we also use it to generate/validate the JS objects for the _gdm protocol_ ([see here](https://salsa.debian.org/gnome-team/gnome-shell/-/blob/ubuntu/noble-authd/debian/patches/ubuntu-authd/Implement-authd-support-via-the-unified-mechanism.patch?ref_type=heads#L734)).

In fact, when I'll have time I would like even to add the authd JS code side here and test with the generated code.

> the naming is inconsistent.

Well could be, this happened earlier on the development when things weren't well organized either and so got moved with the time.

However, I wouldn't block this PR on this either, since that wasn't the scope: I've moved `authd` without renaming it for now, so that it was just an easy move without many refactors, but I think we can discuss how to move all these protocols in another PR, or drop the commit here (but I though that moving it away wasn't too bad since it didn't cause many changes).